### PR TITLE
refactor(core): add /edit and /add routes for address forms

### PR DESCRIPTION
**Pull request by jorgemoya**
## What/Why?

- Splits forms to their own pages (`/add` and `/edit`).
- Updates translations.
- Moves helper functions to shared components.

## Testing
Locally.

### CODE REVIEW

File: core/components/form-fields/shared/parse-fields.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { UpdateCustomerAddressInput } from '~/app/[locale]/(default)/account/(tabs)/addresses/_actions/update-address';
+import { UpdateCustomerAddressInput } from '~/app/[locale]/(default)/account/(tabs)/addresses/edit/[slug]/_actions/update-address';
```
Comment (jorgemoya) [inline core/components/form-fields/shared/parse-fields.ts:1]: Trying to figure out how to get input values without relying on a specific mutation. 🤔

File: core/components/form-fields/shared/field-handlers.ts [hunk 2]
```diff
@@ -225,20 +227,12 @@ const createPicklistOrTextValidationHandler =
     picklistWithTextStateSetter({ ...picklistWithTextState, [fieldId]: !validationStatus });
   };
 
-const createCountryChangeHandler =
-  (provinceSetter: FieldStateSetFn<StateOrProvince>, countries: Countries) => (value: string) => {
-    const states = countries.find(({ code }) => code === value)?.statesOrProvinces;
-
-    provinceSetter(states ?? []);
-  };
-
```
Comment (jorgemoya) [inline core/components/form-fields/shared/field-handlers.ts:234]: This was very specific to the address forms.

File: core/app/[locale]/(default)/account/(tabs)/addresses/edit/[slug]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  const reCaptchaSettings = data.site.settings?.reCaptcha;
+  const countries = data.geography.countries;
+  const addressFields = [...(data.site.settings?.formFields.shippingAddress ?? [])];
+  const addresses = removeEdgesAndNodes({ edges: data.customer?.addresses.edges });
+
+  if (addresses.length === 0) {
+    return notFound();
+  }
+
+  const address = addresses.find((ad) => ad.entityId.toString() === slug);
```
Comment (chanceaclark) [inline core/app/[locale]/(default)/account/(tabs)/addresses/edit/[slug]/page.tsx:125]: 🍹 🧶
```suggestion
  const existingAddress = addresses.find((address) => address.entityId.toString() === slug);
```

File: core/messages/en.json [hunk 4]
```diff
@@ -140,40 +140,45 @@
...
+      },
+      "Edit": {
+        "heading": "Edit address",
+        "Form": {
+          "recaptchaText": "Pass ReCAPTCHA check",
+          "submit": "Edit address",
+          "submitting": "Updating address...",
+          "cancel": "Cancel",
+          "confirmDeleteAddress": "Delete address",
+          "deleteModalTitle": "Are you sure you want to delete address?",
```
Comment (chanceaclark) [inline core/messages/en.json:170]: Maybe add `this`?
```suggestion
          "deleteModalTitle": "Are you sure you want to delete this address?",
```

File: core/app/[locale]/(default)/account/(tabs)/addresses/add/page.tsx [hunk 5]
```diff
@@ -57,7 +59,13 @@ const FALLBACK_COUNTRY = {
   states: [],
 };
 
-export const CustomerNewAddress = async () => {
+export const metadata = {
+  title: 'Add address',
+};
```
Comment (chanceaclark) [inline core/app/[locale]/(default)/account/(tabs)/addresses/add/page.tsx:64]: Should we generate the metadata for these pages instead? Or is there something I am missing around fetching the translations?
```ts
export async function generateMetadata() {
 const t = await getTranslations('Account.Addresses.Add');

  return {
    title: t('heading'),
  }
}
```

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx [hunk 6]
```diff
@@ -1,43 +1,52 @@
...
+};
+
+export default async function Addresses({ searchParams }: Props) {
+  const { before, after } = searchParams;
+
+  const data = await getCustomerAddresses({
     ...(after && { after }),
     ...(before && { before }),
-    limit: action === 'edit-address' ? undefined : 2,
+    limit: 4,
```
Comment (chanceaclark) [inline core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx:28]: 🧶 Can we increase this even more? e.g. 10?

File: core/components/form-fields/shared/parse-fields.ts (hunk 1 above)
Comment (chanceaclark) [reply core/components/form-fields/shared/parse-fields.ts:1]: Could you create a fragment to share across these mutations? Then use the fragment for types?

File: core/app/[locale]/(default)/account/(tabs)/addresses/add/page.tsx (hunk 5 above)
Comment (jorgemoya) [reply core/app/[locale]/(default)/account/(tabs)/addresses/add/page.tsx:64]: Noticed there are a lot of metadatas missing translations, so I will tackle this in a different PR.

File: core/components/form-fields/shared/parse-fields.ts (hunk 1 above)
Comment (jorgemoya) [reply core/components/form-fields/shared/parse-fields.ts]: Created a fake mutation, other solution was to type it myself.

File: core/components/form-fields/shared/parse-fields.ts [hunk 7]
```diff
@@ -1,6 +1,21 @@
-import { UpdateCustomerAddressInput } from '~/app/[locale]/(default)/account/(tabs)/addresses/_actions/update-address';
+import { graphql, VariablesOf } from '~/client/graphql';
+
+// Faking mutation to get input type
+// Not ideal but best way to make sure input type is up to date
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const FakeMutation = graphql(`
```
Comment (chanceaclark) [inline core/components/form-fields/shared/parse-fields.ts:6]: Why don't we just import the type from `~/bigcommerce.graphql.d.ts` instead?

**Review by chanceaclark — APPROVED** [review]

File: core/components/form-fields/shared/parse-fields.ts (hunk 7 above)
Comment (jorgemoya) [reply core/components/form-fields/shared/parse-fields.ts:6]: 👀

Comment (jorgemoya) [reply core/components/form-fields/shared/parse-fields.ts:6]: Will keep as is.

**Pull request merged**